patches/windows/CMakeLists.patch

```diff
@@ -0,0 +1,119 @@
+Index: jni/CMakeLists.txt
+<+>UTF-8
+===================================================================
+Description:
+We are making the following changes to jni/CMakeLists.txt file to build it on Windows OS
+1. Set CXX_COMPILER_VERSION and CMAKE_CXX_FLAGS.
+2. Add a new if else case when the CMAKE_SYSTEM_NAME matches Windows and set flags JVM_OS_TYPE, prefix and extension for the target libraries that are built.
+3. Replace LIBRARY_OUTPUT_DIRECTORY with RUNTIME_OUTPUT_DIRECTORY, to build the target libraries(opensearchknn_common, opensearchknn_nmslib and opensearchknn_faiss) in the
+specified directory at runtime.
+4. Comment the TESTS for now because the tests are failing(failing to build jni_tests.exe) if we are building our target libraries as SHARED libraries.
+
+TODO: Fix the failing JNI TESTS
+
+===================================================================
+diff --git a/jni/CMakeLists.txt b/jni/CMakeLists.txt
+--- a/jni/CMakeLists.txt	(revision 78a2b5b2a83db52aded56ab74940a62c94a88b50)
++++ b/jni/CMakeLists.txt	(date 1666128659873)
+@@ -14,8 +14,10 @@
+ set(TARGET_LIB_FAISS opensearchknn_faiss)    # faiss JNI
+ set(TARGET_LIBS "")  # Libs to be installed
+ 
++set(CXX_COMPILER_VERSION ${CMAKE_CXX_COMPILER_VERSION})
```

Comment:
Patch files are difficult to maintain. Generally, they are better for source files that we don't want to edit (i.e. faiss/nmslib src files). It seems like we should just have branching logic in cmake rather than a patch for cmake.

Reply:
updated